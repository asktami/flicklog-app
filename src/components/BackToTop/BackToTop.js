import React from 'react';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const BackToTop = ({ goBackToTop, showBackToTop, text }) => {
	return (
		<button
			id="back-to-top"
			className="btn back-to-top"
			onClick={goBackToTop}
			style={{ display: showBackToTop ? 'block' : 'none' }}
		>
			back to top
			<br />
			<FontAwesomeIcon icon={['fas', 'chevron-circle-up']} size="3x" /> {text}
		</button>
	);
};

export default BackToTop;
