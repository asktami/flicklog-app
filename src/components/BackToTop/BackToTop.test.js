import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import BackToTop from './BackToTop';

describe(`BackToTop component`, () => {
	it('renders a BackToTop by default', () => {
		const wrapper = shallow(<BackToTop />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
