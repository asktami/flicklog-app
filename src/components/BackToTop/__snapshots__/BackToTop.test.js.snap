// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackToTop component renders a BackToTop by default 1`] = `
<button
  className="btn back-to-top"
  id="back-to-top"
  style={
    Object {
      "display": "none",
    }
  }
>
  back to top
  <br />
  <FontAwesomeIcon
    icon={
      Array [
        "fas",
        "chevron-circle-up",
      ]
    }
    size="3x"
  />
   
</button>
`;
